t/run-tests

```diff
@@ -93,6 +102,16 @@ sub get_logfn {
     "$tempdir/$script.out";
 }
 
+sub print_log {
+    my $script = shift;
+    # print the output of the perl script being run
+    my $logfn = get_logfn($script);
+    open my $logfh, "<", $logfn
+        or die "failed to open script:$logfn:$!";
+    unlink $logfn;
+    print do { local $/; <$logfh> };
+}
```

Comment:
Thank you. Changes to run-tests is a nice improvement while we make other changes in this PR.

How about changing this function to `commit_test` or something, that takes the name of the script and the result as the argument?

I say this because, while we've extract the logging, we still have redundancy regarding how the delimiters (i.e., lines starting from "# $script") and how `@failures` are recorded.

Reply:
@kazuho Thank you for the suggestion, I refactored a bit in https://github.com/h2o/h2o/pull/2925/commits/1b8df40f170b01213ee1666455fd20eba4078c2c.

Reply:
Fantastic! Thank you.